mslib/msui/msui_mainwindow.py

```diff
@@ -1200,23 +1200,50 @@ def closeEvent(self, event):
         if ret == QtWidgets.QMessageBox.Yes:
             if self.mscolab.help_dialog is not None:
                 self.mscolab.help_dialog.close()
-            # cleanup mscolab widgets
             if self.mscolab.token is not None:
                 self.mscolab.logout()
-            # Table View stick around after MainWindow closes - maybe some dangling reference?
-            # This removes them for sure!
+
+            try:
+                # Check listViews for MSUITopViewWindow instances
+                view_windows = []
+                for i in range(self.listViews.count()):
+                    item = self.listViews.item(i)
+                    if item and hasattr(item, 'window') and isinstance(item.window, topview.MSUITopViewWindow):
+                        view_windows.append(item.window)
+
+                if view_windows:
+                    all_settings = []
+                    for view_window in view_windows:
+                        if hasattr(view_window, 'get_settings'):
+                            settings = view_window.get_settings()
+                            if settings and settings.get("view_type") == "topview":
+                                all_settings.append(settings)
+                    if all_settings:
+                        view_restoration.save_view_settings(all_settings)
+                        logging.info("Saved settings for all topview windows on application close")
+                    else:
+                        logging.info("No valid top view settings to save on close.")
+                else:
+                    logging.warning("No top view windows found to save settings.")
+            except Exception as e:
```

Comment:
please look what Exception(s) usually could happen here. It better to catch only those.